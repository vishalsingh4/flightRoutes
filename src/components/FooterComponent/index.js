// css import

import './index.scss';

const FooterComponent = () => {
  return (
    <div className='footer-wrapper bg-dark color-white fs-15 fw-600 d-flex align-items-center justify-content-between p-3'>
      <ul className='footer-list d-flex justify-content-around align-items-center w-50 p-0 m-0'>
        <li className='link' tabIndex='0' aria-label='about us'>ABOUT US</li>
        <li className='link' tabIndex='0' aria-label='privacy'>PRIVACY</li>
        <li className='link' tabIndex='0' aria-label='terms of use'>TERMS OF USE</li>
        <li className='link' tabIndex='0' aria-label='careers'>CAREERS</li>
        <li className='link' tabIndex='0' aria-label='customer service'>CUSTOMER SERVICE</li>
      </ul>
      <ul className='p-0 m-0'>
        <li className='copyright' tabIndex='0' aria-label='copyright Le Travenues Technology Ltd. Dubai'>&copy; 2021 Le Travenues Technology Ltd. Dubai</li>
      </ul>
    </div>
  )
};

export default FooterComponent;
