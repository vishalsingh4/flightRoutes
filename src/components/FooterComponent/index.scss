  /* Extra small devices (phones, 600px and down) */
  @media only screen and (max-width: 600px) {
    .footer-section {
       .footer-wrapper {
        font-size: 7px;
        flex-direction: column;
        padding: 10px !important;

        .footer-list {
            width: 100% !important;
        }
       }
    }
  }

/* Small devices (portrait tablets and large phones, 600px and up) */
  @media only screen and (min-width: 600px) and (max-width: 768px){
    .footer-section{
      .footer-wrapper {
        font-size: 10px;
        flex-direction: column;
        padding: 12px !important;

        .footer-list {
            width: 100% !important;
        }
       }
    }
  }

/* Medium devices (landscape tablets, 768px and up) */
  @media only screen and (min-width: 768px) and (max-width: 992px){
    .footer-section{
      .footer-wrapper {
        font-size: 15px;
        flex-direction: column;
        padding: 15px !important;

        .footer-list {
            width: 100% !important;
        }
       }
    }
  } 