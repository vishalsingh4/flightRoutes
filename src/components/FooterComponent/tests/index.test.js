import React from 'react';
import { shallow } from 'enzyme';
import FooterComponent from '..';

describe("Footer Component", () => {
    const wrapper = shallow(<FooterComponent />);

    it('should render the Footer Component correctly - snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('renders component', () => {
        expect(wrapper.exists()).toBe(true);
    });
    it('renders the links', () => {
        expect(wrapper.find('.link').at(0).text()).toEqual('ABOUT US');
        expect(wrapper.find('.link').at(1).text()).toEqual('PRIVACY');
        expect(wrapper.find('.link').at(2).text()).toEqual('TERMS OF USE');
        expect(wrapper.find('.link').at(3).text()).toEqual('CAREERS');
        expect(wrapper.find('.link').at(4).text()).toEqual('CUSTOMER SERVICE');
    });
    it('render copyright', () => {
        expect(wrapper.find('.copyright').text()).toEqual('Â© 2021 Le Travenues Technology Ltd. Dubai');
    });
})