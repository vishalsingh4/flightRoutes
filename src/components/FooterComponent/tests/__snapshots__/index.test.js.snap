// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component should render the Footer Component correctly - snapshot 1`] = `
<div
  className="footer-wrapper bg-dark color-white fs-15 fw-600 d-flex align-items-center justify-content-between p-3"
>
  <ul
    className="footer-list d-flex justify-content-around align-items-center w-50 p-0 m-0"
  >
    <li
      aria-label="about us"
      className="link"
      tabIndex="0"
    >
      ABOUT US
    </li>
    <li
      aria-label="privacy"
      className="link"
      tabIndex="0"
    >
      PRIVACY
    </li>
    <li
      aria-label="terms of use"
      className="link"
      tabIndex="0"
    >
      TERMS OF USE
    </li>
    <li
      aria-label="careers"
      className="link"
      tabIndex="0"
    >
      CAREERS
    </li>
    <li
      aria-label="customer service"
      className="link"
      tabIndex="0"
    >
      CUSTOMER SERVICE
    </li>
  </ul>
  <ul
    className="p-0 m-0"
  >
    <li
      aria-label="copyright Le Travenues Technology Ltd. Dubai"
      className="copyright"
      tabIndex="0"
    >
      Â© 2021 Le Travenues Technology Ltd. Dubai
    </li>
  </ul>
</div>
`;
